### ISSUE TITLE
Crash in release mode

### ISSUE THREAD

**Issue by zsajjad** [Agontuk/react-native-geolocation-service#11]
Our crashlytics console is showing this crash in production app.

```
Fatal Exception: java.lang.NullPointerException
...
com.google.android.gms.common.api.Status.startResolutionForResult (Unknown Source)
com.google.android.gms.common.api.ResolvableApiException.startResolutionForResult (Unknown Source)
com.agontuk.RNFusedLocation.RNFusedLocationModule.onLocationSettingsResponse (RNFusedLocationModule.java:173)
com.agontuk.RNFusedLocation.RNFusedLocationModule.access$200 (RNFusedLocationModule.java:31)
com.agontuk.RNFusedLocation.RNFusedLocationModule$2.onComplete (RNFusedLocationModule.java:126)
...
```

**Comment by Agontuk** [comment]
Can you provide the full log, along with which play service version is being used, android version etc ?

**Comment by zsajjad** [comment]
Thanks for the library and special thanks for your prompt response.

build.gradle looks like
```
    implementation 'com.google.android.gms:play-services-location:11.8.0'
...
    implementation (project(':react-native-geolocation-service')) {
        exclude group: 'com.google.android.gms', module: 'play-services-location'
    }
```

### About Devices
<img width="594" alt="screen shot 2018-03-02 at 12 32 50 am" src="https://user-images.githubusercontent.com/15834890/36865418-46f14394-1db1-11e8-9f21-420f870d1d7f.png">
<img width="365" alt="screen shot 2018-03-02 at 12 33 34 am" src="https://user-images.githubusercontent.com/15834890/36865442-5e2b77dc-1db1-11e8-9214-67d654af34d5.png">

### Complete Logs
<img width="944" alt="screen shot 2018-03-02 at 12 34 20 am" src="https://user-images.githubusercontent.com/15834890/36865482-7b399930-1db1-11e8-899e-42b3503d4671.png">

**Comment by Agontuk** [comment]
Is this a consistent crash in release mode or only happening once or twice ?

**Comment by zsajjad** [comment]
This issue has 6 crashes affecting 5 users out of 500 beta testers.

**Comment by Agontuk** [comment]
Are you using proguard ?

**Comment by zsajjad** [comment]
Yes we are using proguard.

**Comment by zsajjad** [comment]
Will try to reproduce it at our end.

**Comment by Agontuk** [comment]
About the `unknown source`, you should setup crashlytics for proguard with the maaping.txt file generated during build. Then you'll get a proper crash log.

**Comment by Agontuk** [comment]
This should be fixed in `1.0.3`. CurrentActivity can sometimes be null, in that case it'll now invoke the error callback with code -1.

**Comment by zsajjad** [comment]
Hey, Thanks a lot man.
